src/sql/parts/notebook/cellViews/code.component.ts

```diff
@@ -135,6 +136,23 @@ export class CodeComponent extends AngularDisposable implements OnInit, OnChange
 		}
 	}
 
+	private updateConnectionState(isConnected: boolean) {
+		if (this.isSqlCodeCell()) {
+			let cellUri = this.cellModel.cellUri.toString();
+			if (!isConnected && this._model.notebookOptions.connectionService.isConnected(cellUri)) {
+				this._model.notebookOptions.connectionService.disconnect(cellUri).catch(e => console.log(e));
+			} else if (this._model.activeConnection && this._model.activeConnection.id !== '-1') {
```

Comment:
Just checking -  there is no scenario where connectionService is connected and connection id == -1 ? In that case we might need to add (this._model.activeConnection.id === '-1' || !isConnected) in If condition.

Reply:
Not that I know of. It starts at 0. We use this logic in other places so it seems a sensible way to identify invalid connections without special extra logic.